R/pkg/R/DataFrame.R

```diff
@@ -55,6 +55,19 @@ setMethod("initialize", "SparkDataFrame", function(.Object, sdf, isCached) {
   .Object
 })
 
+#' Set options/mode and then return the write object
+#' @noRd
+setWriteOptions <- function(write, path = NULL, mode = 'error', ...) {
+    options <- varargsToStrEnv(...)
+    if (!is.null(path)) {
+      options[["path"]] <- path
+    }
+    jmode <- convertToJSaveMode(mode)
+    write <- callJMethod(write, "mode", jmode)
+    write <- callJMethod(write, "options", options)
+    write
```

Comment:
Yes, I think that sounds good. Actually, I was thinking single common method for options in both `read`/`write` (maybe in `utils.R`?) and two common method for reading/writing in `read`/`write`. I am wondering maybe if you think it is okay for me to try this in another PR after this one/https://github.com/apache/spark/pull/15231